.github/workflows/ci.yml

```diff
@@ -13,41 +13,28 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest]
+        ocaml-compiler:
+          - 4.14.x
     steps:
-      - uses: actions/setup-node@v1
-        with:
-          node-version: 12
+      - name: Checkout code
+        uses: actions/checkout@v2
 
-      - uses: actions/checkout@v2
-      - name: Cache esy dependencies
-        uses: actions/cache@v1
-        id: cache
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
         with:
-          path: _export
-          key: ${{ runner.OS }}-build-${{ hashFiles('esy.lock/index.json') }}
-          restore-keys: |
-            ${{ runner.OS }}-build-${{ env.cache-name }}-
-            ${{ runner.OS }}-build-
-            ${{ runner.OS }}-
-
-      - name: Install esy
-        run: npm install -g esy@latest
-
-      - name: Import dependencies
-        if: steps.cache.outputs.cache-hit == 'true'
-        run: esy import-build _export/*
-
-      - name: Install dependencies
-        run: esy install
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          dune-cache: ${{ matrix.os != 'macos-latest' }}
 
-      - name: Build
-        run: esy build --release
+      - name: Install opam packages
+        run: opam install . --with-test --locked
 
       - name: Check formatting
-        run: esy format
+        run: opam exec -- dune build @fmt
+        if: ${{ matrix.os == 'ubuntu-latest' && always() }}
 
-      - name: Run tests
-        run: esy test
+      - name: Run build
+        run: opam exec -- dune build
 
-      - name: Export dependencies
-        run: esy export-dependencies
+      - name: Run the unit tests
```

Comment:
IMO we should consider running tests in docker, it'll also provide faster builds because the constant layers (the dependencies) will be skipped. What do you think?

Reply:
yes we could, I wanted to check how fast CI was with cache before making such decisions. But it'd make sense

Reply:
Im worried about the dev container performance on Mac.

Reply:
My experiences so far weren't great perf wise but I only tried java so YMMV